Annotations/Classes/Views/ObfuscateCanvas.swift

```diff
@@ -6,26 +6,28 @@ protocol ObfuscateCanvas: class, ObfuscateViewDelegate {
 }
 
 extension ObfuscateCanvas {
-  func redrawObfuscates(model: CanvasModel) {
-    for (index, model) in model.obfuscates.enumerated() {
-      let state = ObfuscateViewState(model: model, isSelected: false)
-      let view = ObfuscateViewClass(state: state, modelIndex: index, color: model.color)
-      view.delegate = self
-      add(view)
-    }
+  func redrawObfuscate(model: ObfuscateModel, canvas: CanvasModel) {
+    guard let modelIndex = canvas.obfuscates.firstIndex(of: model) else { return }
+    let state = ObfuscateViewState(model: model, isSelected: false)
+    let view = ObfuscateViewClass(state: state, modelIndex: modelIndex,
+                                  globalIndex: model.index, color: model.color)
+    view.delegate = self
+    add(view)
   }
   
   func createObfuscateView(origin: PointModel, to: PointModel, color: ModelColor) -> (CanvasDrawable?, KnobView?) {
     if origin.distanceTo(to) < 5 {
       return (nil, nil)
     }
     
-    let newRect = RectModel(origin: origin, to: to, color: color)
+    let newRect = ObfuscateModel(index: model.elements.count + 1,
+                            origin: origin, to: to, color: color)
```

Comment:
@dworak it would be better to start each parameter from a new line here for better readability